src/main/java/pro/trevor/tankgame/rule/impl/version4/Ruleset.java

```diff
@@ -1,7 +1,32 @@
 package pro.trevor.tankgame.rule.impl.version4;
 
+import pro.trevor.tankgame.rule.definition.RulesetDescription;
+import pro.trevor.tankgame.rule.definition.player.PlayerRuleset;
 import pro.trevor.tankgame.rule.impl.IRuleset;
+import pro.trevor.tankgame.rule.impl.shared.PlayerRules;
 import pro.trevor.tankgame.rule.impl.util.BaseRuleset;
+import pro.trevor.tankgame.rule.impl.version3.Tank;
+import pro.trevor.tankgame.util.LineOfSight;
 
 public class Ruleset extends BaseRuleset implements IRuleset {
+
+    @Override
+    public void registerPlayerRules(RulesetDescription ruleset) {
+        PlayerRuleset playerRules = ruleset.getPlayerRules();
+
+        playerRules.put(Tank.class, PlayerRules.SpendActionToShootWithDeathHandle(LineOfSight::hasLineOfSightV4,
+            (s, t, d) -> {
+                t.setGold(t.getGold() + d.getBounty());
+                switch (d.getGold()) {
+                    case 0 -> {}
+                    case 1 -> t.setGold(t.getGold() + 1);
+                    default -> {
+                        // Tax is target tank gold * 0.25 rounded up
+                        int tax = (d.getGold() + 3) / 4;
```

Comment:
This logic is pretty complex. I'm going to review it some more to make sure it's right, but you need to add some unit tests for this as well. Focus on the looting aspect, but really all parts of shooting should be handled. This is also why I moved all rule logic into functions or static final variables, so they could be fully unit tested. Right now, with this logic being in the ruleset, it cannot be tested.

Reply:
Tested and DONE